### PR TITLE
Backend part of legacy api management

### PR DESCRIPTION
**Pull request by nanaya**
That's rather large.

- [x] depends on #10111

### CODE REVIEW

File: app/Models/ApiKey.php [hunk 1]
```diff
@@ -18,7 +25,72 @@
  */
 class ApiKey extends Model
```
Comment (notbakaneko) [inline app/Models/ApiKey.php:26]: could do with including `* @property-read User|null $user`

File: app/Models/User.php [hunk 2]
```diff
@@ -42,7 +42,7 @@
 
 /**
  * @property-read Collection<UserAccountHistory> $accountHistories
- * @property-read ApiKey|null $apiKey
+ * @property-read Collection<ApiKeys>|null $apiKeys
```
Comment (notbakaneko) [inline app/Models/User.php:45]:
```suggestion
 * @property-read Collection<ApiKey> $apiKeys
```

**Review by notbakaneko — CHANGES_REQUESTED** [review]

**Review by notbakaneko — APPROVED** [review]
still need a UI for this 👀

**Comment by nanaya** [comment]
These prs are intended to be merged first btw 👀

**Pull request merged**